vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3209,13 +3209,49 @@ public void setMultiSort(boolean multiSort) {
      * @see MultiSortPriority
      */
     public void setMultiSort(boolean multiSort, MultiSortPriority priority) {
-        Objects.requireNonNull(priority,
-                "Multi-sort priority must not be null");
         setMultiSort(multiSort);
         updateMultiSortPriority(priority);
     }
 
+    /**
+     * Sets whether multiple column sorting is enabled on the client-side.
+     *
+     * @param multiSort
+     *            {@code true} to enable sorting of multiple columns on the
+     *            client-side, {@code false} to disable
+     * @param onShiftClickOnly
+     *            {@code true} to enable multi-sort by shift-clicking,
+     *            {@code false} for normal multi-sort behavior
+     */
+    public void setMultiSort(boolean multiSort, boolean onShiftClickOnly) {
+        getElement().setAttribute("multi-sort-on-shift-click",
+                multiSort && onShiftClickOnly);
```

Comment:
Is the check for multiSort necessary here? Could it just be `getElement().setAttribute("multi-sort-on-shift-click", onShiftClickOnly);`?

It would be preferable to operate directly with the properties instead of attributes (which get parsed as properties by Polymer): `getElement().setProperty("multiSortOnShiftClick", onShiftClickOnly);`

Reply:
> Is the check for multiSort necessary here? Could it just be getElement().setAttribute("multi-sort-on-shift-click", onShiftClickOnly);?
I am not sure about this to be honest. My thinking is that it would cause confusion if the user is able to `setMultiSort(false, true)`, thinking they changed multi-sort to work on shift-click only but they have disabled multi-sort. If that makes sense. Actually I will add extra documentation to clarify this and it should solve any problems.

Reply:
Fair enough. My reasoning for this was that on the Web Component: `<vaadin-grid multi-sort-on-shift-click>` also works even without adding `multi-sort`

Reply:
Removed the extra check, added more documentation and refactored to use property instead of attribute.